'Fixed' eslint issue.

tejas-hosamani opened the pull request:
This wasn't really much of a fix. 
Just tried a bunch of things only to notice `"printWidth": 200`, later. Changin that value to 100 made wrapping work as expected. 

Further, I've noticed 
```
"vetur.format.defaultFormatter.html": "prettyhtml" // VS Code setting
```
and 
```
"vetur.format.defaultFormatter.html": "prettier" // VS Code setting
```
had no effect on formatting. So, project settings are independent of VSCode settings.
However, Vetur recommends `prettyhtml` for that particular setting. I am using it currently.

tejas-hosamani committed:
'Fixed' eslint issue.

spiray commented on .prettierignore:
```diff
@@ -0,0 +1 @@
+src/components/HelloWorld.vue
```
Can we just get rid of this file?

tejas-hosamani replied:
Yes, I 'll update that separately. Need to write some dummy test or something.

spiray commented on .prettierrc:
```diff
@@ -1,5 +1,5 @@
 {
-  "printWidth": 200,
+  "printWidth": 100,
```
This seems like a hacky way to make the vue attributes to wrap. When you only have 2 that you want to stay on separate lines this may not work. Does that make sense?

tejas-hosamani replied:
I am not sure what you mean. Could you please explain?

tejas-hosamani replied:
Thoughts on this? I am thinking this is the proper way to make attributes wrap if they exceed a certain length(?) I can't think of any other way.

spiray replied:
My thought on this was that if you have an element with 3 attributes that you would want to wrap but the line length is less than 100 it wont work in that case. 100 is kind of an arbitrary number...?

tejas-hosamani replied:
Yes, 100 is an arbitrary number. And yes, it wouldn't wrap if there are three attributes and it doesn't exceed 100 chars. I think the essential point is to make code readable. If there are three attributes and they do not exceed 100 chars, it's probably still readable. but I now get what you meant by hacky way. It is so, yes.

I wonder if there is a way to actually wrap attributes if they are more than 2, that would be neat. I 'll try looking for that.

Thanks for clarifying 🙂

tejas-hosamani replied:
**Found it!**

This is the rule: https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/max-attributes-per-line.md

It's set to 1 attribute by default. I think that's good.

This rule automatically comes with `vue/strongly-recommended` and `vue/recommended` plugins.

Currently, we are using `vue/essential`. I think we should shift to one of those.

It's just a matter of changing the text `essential` to `recommended`, for example.

What do you think?
@spiray

spiray replied:
That sounds good, Can you see if it will introduce other issues?

tejas-hosamani replied:
Yup. I am testing it right now. I am seeing some weird behavior. But it's still under control :)

tejas-hosamani replied:
@spiray Nope. Eslint and Prettier have a conflict of interest, to put it mildly.

What we have currently seems to be practically best.

spiray commented on src/registerServiceWorker.js:
```diff
@@ -5,7 +5,9 @@ import { register } from 'register-service-worker';
   register(`${process.env.BASE_URL}service-worker.js`, {
     ready() {
-      console.log('App is being served from cache by a service worker.\nFor more details, visit https://goo.gl/AFskqB');
+      console.log(
```
This is a downside to making the printWidth 100.

tejas-hosamani replied:
Yes. but we could ignore this file if we want to. On other cases, the code isn't readable with 200 width. Such lines would require to scroll through horizontally. And again, hard to read through attributes and stuff. What do you think?

tejas-hosamani replied:
I am updating with an above-mentioned rule but I still think 100 width is good to maintain readability. And I propose to add this file to `.prettierignore`. Because this is the only file we have this problem with. What do you think?

spiray replied:
I generally use 120 but it doesn't really matter

tejas-hosamani replied:
Ah. I think our preferences come from our screen sizes. Do you happen to have a really big screen? Just curious.

spiray replied:
I have a 24" desktop but my laptop is 15.6". Are you on 13"?

tejas-hosamani replied:
No, I've got the same size monitors too. Hmm. Anyway. 

Ah, I use a bit of zoom on my editor. Maybe that's the reason 🙂

tejas-hosamani committed:
Eslint fix proposal

tejas-hosamani commented:
Alright. So this is the best I could come up with.

* No prettier onSave action on code editor. NONE AT ALL. This messes things up.
* Linting runs in the background when we have a local server running - which shows all the lint errors and warnings as we make changes - this is a lot better compared to what we had previously, it would ONLY lint during pre-commit hook trigger.
* ~~Using `'plugin: prettier/recommended'` instead of we defining them. So, essentially, .prettierrc is pretty much useless.~~

tejas-hosamani committed:
Added .prettierrc back.

tejas-hosamani committed:
Added .prettierrc back.

tejas-hosamani committed:
Added .prettierrc back.

tejas-hosamani commented:
Essentially, this is whatever it was before + dynamic linting. Meaning, lint runs in the background and shows all the errors and warning as long as local server keeps running so that we don't have to worry about unexpected pre-commit lint-fixes. Not the best but a little better.

tejas-hosamani committed:
Removed merge conflicts.

tejas-hosamani committed:
Minor update.

tejas-hosamani committed:
Minor update.

tejas-hosamani committed:
Minor update.

spiray reviewed (approved):
The linting side looks good. @rabbitwerks Can you review the Vue files?

tejas-hosamani commented:
Those are actually already merged changes. Just applied lint to them, that's all. :)

spiray commented:
I see, let's wait for one more review

tejas-hosamani commented:
Just to document this PR for anyone going through it in the future:

The major changes here are:
1. Added eslint rule
	```
	'vue/max-attributes-per-line': [
	      'error',
	      {
	        singleline: 2,
	        multiline: {
	          max: 1,
	          allowFirstLine: false,
	        },
	      },
	    ],
	```
	Please find more details about this rule [here](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/max-attributes-per-line.md).

1. Added `vue.config.js` file.
	In this instance, this file(though it's empty), enables `lintOnSave`. Which helps us avoid worrying about unexpected lint-fixes during pre-commit hook runtime.
	Please find more details about this file [here](https://cli.vuejs.org/config/#vue-config-js).

tejas-hosamani commented:
@ewenjo could you please review this when you are able..
